lib/Event/ScheduleCriteriaRequestInterface.php

```diff
@@ -0,0 +1,68 @@
+<?php
+/*
+ * Copyright (C) 2024 Xibo Signage Ltd
+ *
+ * Xibo - Digital Signage - https://xibosignage.com
+ *
+ * This file is part of Xibo.
+ *
+ * Xibo is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * any later version.
+ *
+ * Xibo is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Xibo.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace Xibo\Event;
+
+use Xibo\Support\Exception\ConfigurationException;
+
+/**
+ * Interface for managing schedule criteria types, metrics, and values.
+ *
+ * Allows the addition of types, metrics, and values in a chained manner:
+ * - Start with addType() to add a new type. Call addType() multiple times to add multiple types.
+ * - Follow with addMetric() to add metrics under the specified type. Call addMetric() multiple times to add multiple metrics to the current type.
```

Comment:
phpcs error on line length - do you have your IDE configured to show you these in real time?

Reply:
I'm sorry, I overlooked the exceeded line length. But, its already configured in my IDE. I'll keep this in mind moving forward.